File: 001.js
// JavaScript bank application API

let accountBalance = 1000;

function getAccountBalance() {
  return accountBalance;
}

function deposit(amount) {
  if (amount <= 0) {
    throw new Error("Invalid amount. Amount must be greater than zero.");
  }
  accountBalance += amount;
}

function withdraw(amount) {
  if (amount <= 0) {
    throw new Error("Invalid amount. Amount must be greater than zero.");
  }
  if (amount > accountBalance) {
    throw new Error("Insufficient balance.");
  }
  accountBalance -= amount;
}

File: 001.cs
// C# bank application API

public static class BankApplicationAPI {
    private static double accountBalance = 1000;

    public static double GetAccountBalance() {
        return accountBalance;
    }

    public static void Deposit(double amount) {
        if (amount <= 0) {
            throw new ArgumentException("Invalid amount. Amount must be greater than zero.");
        }
        accountBalance += amount;
    }

    public static bool Withdraw(double amount) {
        if (amount <= 0) {
            throw new ArgumentException("Invalid amount. Amount must be greater than zero.");
        }
        if (amount > accountBalance) {
            throw new InvalidOperationException("Insufficient balance.");
        }
        accountBalance -= amount;
        return true;
    }
}

